internal_workflow_testsuite.go

```diff
@@ -343,8 +341,10 @@ func (env *testWorkflowEnvironmentImpl) executeActivity(
 		input,
 	)
 
-	// ensure activityFn is registered to defaultTestTaskList
-	env.testSuite.RegisterActivity(activityFn)
+	if _, ok := getHostEnvironment().getActivityFn(fnName); !ok {
```

Comment:
If we remove the Register* method from test suite, how can you make use of the private hostEnv?